DEVELOPERS_README.dox

```diff
@@ -0,0 +1,180 @@
+ * votca::tools. This object only has a header file it is in:
+ * /tools/include/votca/tools/vec.h
+ * 
+ * Determine if a tests folder has already been created or not in /src if it
+ * has we will skip to step 5
+ *
+ * 1. Create a new folder in src/ called tests if it does not exist
+ * 2. Edit the tools/src/CmakeLists.txt file to include the tests folder:
+ *    \code{.unparsed}
+add_subdirectory(tests)
+\endcode
+ *
+ * 3. Edit tools/the CMakeLists.txt file by ensuring the unit_test_framework is
+ *    included for boost. There should alread exist a line like this in the file:
+ *    \code{.unparsed} 
+find_package(Boost 1.39.0 REQUIRED COMPONENTS program_options filesystem system ) 
+\endcode
+ *    Simply add the unit_test_framework keyword to the list of required components
+ *
+ * 4. Create a CMakeLists.txt file in tools/src/tests/ with the following contents
+ *    \code{.unparsed} 
+if(ENABLE_TESTING)
+  foreach(PROG test_vec )
+    ADD_DEFINITIONS(-DBOOST_TEST_DYN_LINK)
+    file(GLOB ${PROG}_SOURCES ${PROG}*.cc)
+    add_executable(unit_${PROG} ${${PROG}_SOURCES})
+    target_link_libraries(unit_${PROG} votca_tools)
+
+    add_test(unit_${PROG} unit_${PROG})
+    # run tests for tools (for coverage) as well
+    set_tests_properties(unit_${PROG} PROPERTIES LABELS "tools;votca")
+  endforeach(PROG)
+endif(ENABLE_TESTING)
+\endcode
+ * 5. Create a test file in tools/src/tests/test_vec.cc must have the same name as
+ *    what appears in the foreach in the CMakeLists.txt file. And place the following 
+ *    contents
+ *    \code{.cpp}  
+#define BOOST_TEST_MAIN
+
+#define BOOST_TEST_MODULE vec_test
+#include <boost/test/unit_test.hpp>
+#include <exception>
+
+#include <votca/tools/vec.h>
+
+using namespace std;
+using namespace votca::tools;
+
+BOOST_AUTO_TEST_SUITE(vec_test)
+
+
+BOOST_AUTO_TEST_CASE(test1){
+  vecv;
+  BOOST_CHECK_EQUAL(...);
+  BOOST_CHECK_EQUAL(...);
+:
+}
+BOOST_AUTO_TEST_CASE(test2){
+  vecv;
+  BOOST_CHECK_EQUAL(...);
+  BOOST_CHECK_EQUAL(...);
+:
+}
+:
+BOOST_AUTO_TEST_SUITE_END()
+\endcode
+ *    Replace the '...' and ':' with the appropriate syntax. 
+ *    For more info on which boost test macros to use refer to the boost documentation
+ * 6. To compile and test the code create a folder tools/build and run the following
+ *    commands: 
+ *    \code{.unparsed}
+cmake -DENABLE_TESTING=ON ../
+make 
+make test
+\endcode
+ *    Ensure you have an up to date version of cmake or use cmake3 
+ *
+ * \section CPP Codeing Style Guide
+ *
+ * VOTCA now uses the clang formatter to automatically ensure consistent style.
+ * The style is essentially the same, as what is used by google, with the 
+ * exception that types and = are aligned to make the code more readable. 
+ *
+ * To run clang on a file say file.cc I need to first be sure that the 
+ * .clang-format file is placed in the current working directory. It should
+ * have the following contents:
+ * \code{.unparsed}
```

Comment:
Meh, just put this in `.clang-format` in the main folder and refer to that. The `.clang-format` is also automatically pick up by clang-format.